File: problems/3 - Largest prime factor.py
'''The prime factors of 13195 are 5, 7, 13 and 29.
What is the largest prime factor of the number 600851475143?'''


number = 600851475143
divisor = 2

while number != 1:
    if number % divisor == 0:
        print('Number:', number)
        number /= divisor
        print('Factor prime:', divisor)
    else:
        divisor += 1

File: c/3_largest_prime_factor.c
#include<stdlib.h>

int main() {
    long long number = 600851475143LL;
    int divisor = 2;

    while (number != 1) {
        if (number % divisor == 0) {
            number /= divisor;
        } else {
            divisor += 1;
        }
    }

    printf("%d", divisor);

    return 0;
}
